app/utils/results.py

```diff
@@ -223,3 +223,90 @@ def add_ip_card(html_soup: BeautifulSoup, ip: str) -> BeautifulSoup:
         # Inserting the element
         ref_element.insert_before(ip_tag)
     return html_soup
+
+
+def check_currency(response: str):
+    """Check whether the results have currency conversion
+
+    Args:
+        response: Search query Result
+
+    Returns:
+        list
+
+    """
+    soup = BeautifulSoup(response, 'html.parser')
+    a = soup.find(class_='ZINbbc xpd O9g5cc uUPGi')
+    tag = a.find('a')
+    if tag and tag['href'] == 'https://g.co/gfd':
+        currency1 = soup.find(class_="xUrNXd UMOHqf")
+        currency2 = soup.select_one('.kCrYT'
+                                    ' .BNeawe.iBp4i.AP7Wnd '
+                                    '.BNeawe.iBp4i.AP7Wnd')
```

Comment:
Relying on this many obfuscated classes from Google makes me a bit nervous, since they could change any of these and kill this feature as a result...but I'm not sure of a better alternative. 

Could you maybe try finding the `a` tag with the link related to currency, using something like:

```py
currency_link = soup.find('a', {'href': 'https://g.co/gfd'})
```

and then maybe if that's found, traverse up the `div` tree using `<element>.parent` (swapping `<element>` for the previous parent result) until one has the `ZINbbc` class? `ZINbbc` seems to be the main class that Google uses for result divs, so if/when they change that one, it would be an easy thing to replace with whatever they change it to.

Reply:
Yes. I will try to do so. Don't know why I took the harder route 😅. Well maybe I could ensure that the text present is actually just disclaimer?  Instead of checking for parent?